assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js

```diff
@@ -87,12 +88,11 @@ function useUploadMedia({ media, pagingNum, mediaType, fetchMedia, setMedia }) {
             return { localResource, file, element };
           });
         }
-        setMedia({
-          media: [
-            ...localFiles.map(({ localResource }) => localResource),
-            ...media,
-          ],
-        });
+        updatedMedia = [
```

Comment:
Would it make sense to have instead a targeted method `prependMedia` and resolve array logic and stuff on the reducer level?

Reply:
Definitely, that would be a much better way. I think this should be done in a separate PR.